_proto/hapi/release/hook.proto

```diff
@@ -0,0 +1,39 @@
+// Copyright 2016 The Kubernetes Authors All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package hapi.release;
+
+option go_package = "release";
+
+// Hooks describes all of the hooks Tiller recognizes.
+message Hooks {
+	repeated hapi.release.Hook pre_install = 1;
+	repeated hapi.release.Hook post_install = 2;
+	repeated hapi.release.Hook pre_delete = 3;
+	repeated hapi.release.Hook post_delete = 4;
+	repeated hapi.release.Hook pre_upgrade = 5;
+	repeated hapi.release.Hook post_upgrade = 6;
+}
+
+// Hook defines a hook object.
+message Hook {
+	// Kind is the Kubernetes kind.
+	string kind = 1;
+	// Path is the chart-relative path to the template.
+	string path = 2;
+	// Manifest is the manifest contents.
+	string manifest = 3;
+}
```

Comment:
Seems good to me. I'm not familiar with the constraints imposed by parsing yaml to answer this myself, but aside from adding a 'name' field to hook, any reason why you don't define a 'event' or 'action' field instead of using the repeated fields?

i.e.:
```

message Hooks {
	repeated hapi.release.Hook hook = 1;
}

// Hook defines a hook object.
 message Hook {
	// Name is the name of the hook.
	string name = 1;
 	// Kind is the Kubernetes kind.
 	string kind = 2;
 	// Path is the chart-relative path to the template.
 	string path = 3;
 	// Manifest is the manifest contents.
 	string manifest = 4;

	// Action specifies the event trigger of the hook.
	message Action {
		enum Event {
			UNKNOWN  = 0;
			DELETE   = 1;
			INSTALL  = 2;
			UPGRADE  = 3;
		}

		bool pre    = 1;
		bool post   = 2;
		Event event = 3;   
	}

	repeated Action actions = 5;
 }
```

Otherwise, this LGTM.